### PR TITLE
fix: add decision step between header sync and pruned/archival

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---

- removes pruned metadata sync logic from listening state
- adds `DecideNextSync` step that is responsible for finding a suitable peer for block/pruned sync
- chain metadata service includes peer latency
- DRY'd up chain metadata service ping/pong event handling
- take latency into account when selecting a sync peer   
- header sync orders the peers in ascending latency order

Motivation and Context
---
When transitioning to header sync, the listening state would filter out peers that have a suitable horizon height.
However any peer regardless of pruned height can provide headers for header sync.

How Has This Been Tested?
---
Sync tests
Manually

### CODE REVIEW

File: base_layer/core/src/base_node/state_machine_service/states/sync_decide.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+/// Find the peer with the best latency
+fn find_best_latency<'a, I: IntoIterator<Item = &'a SyncPeer>>(iter: I) -> Option<SyncPeer> {
+    iter.into_iter()
+        .fold(Option::<&'a SyncPeer>::None, |current, sync_peer| match current {
+            Some(p) => match (p.latency(), sync_peer.latency()) {
+                (Some(_), None) => Some(p),
+                (None, Some(_)) => Some(sync_peer),
+                (Some(current), Some(l)) if current > l => Some(sync_peer),
```
Comment (stringhandler) [inline base_layer/core/src/base_node/state_machine_service/states/sync_decide.rs]: is this correct?

Comment (sdbondi) [reply base_layer/core/src/base_node/state_machine_service/states/sync_decide.rs]: Maybe can name those better, "if current peer's latency is worse (more) than this peer's latency, use this peer"

Comment (stringhandler) [reply base_layer/core/src/base_node/state_machine_service/states/sync_decide.rs]: was just wondering about the `1` value, I expected `current > sync_peer` or is that a lower case L ?

Comment (sdbondi) [reply base_layer/core/src/base_node/state_machine_service/states/sync_decide.rs]: lol, yeah L for latency, I'll change it - also see there is a test failing - will fix

**Review by stringhandler — APPROVED** [review]

**Pull request merged**